providers/tests/system/google/cloud/cloud_sql/example_cloud_sql_query_iam.py

```diff
@@ -0,0 +1,491 @@
+                    fields="ipAddresses",
+                )
+                response = request.execute()
+                for ip_item in response.get("ipAddresses", []):
+                    if run_in_composer():
+                        if ip_item["type"] == "PRIVATE":
+                            return ip_item["ipAddress"]
+                    else:
+                        if ip_item["type"] == "PRIMARY":
+                            return ip_item["ipAddress"]
+                return None
+
+        get_ip_address_task = get_ip_address(instance=cloud_sql_instance_name)
+
+        conn_id = f"{CONNECTION_PUBLIC_TCP_SSL_ID}_{database_type}"
+
+        @task(task_id=f"create_connection_{database_type}")
+        def create_connection(
+            connection_id: str, instance: str, db_type: str, ip_address: str, port: str
+        ) -> str | None:
+            session = settings.Session()
+            log.info("Removing connection %s if it exists", connection_id)
+            query = session.query(Connection).filter(Connection.conn_id == connection_id)
+            query.delete()
+
+            connection: dict[str, Any] = deepcopy(CONNECTION_PUBLIC_TCP_SSL_KWARGS)
+            connection["extra"]["instance"] = instance
+            connection["host"] = ip_address
+            connection["extra"]["database_type"] = db_type
+            connection["port"] = port
+            conn = Connection(conn_id=connection_id, **connection)
+            session.add(conn)
+            session.commit()
+            log.info("Connection created: '%s'", connection_id)
+            return connection_id
+
+        create_connection_task = create_connection(
+            connection_id=conn_id,
+            instance=cloud_sql_instance_name,
+            db_type=database_type,
+            ip_address=get_ip_address_task,
+            port=db_provider["port"],
+        )
+
+        @task(task_id=f"create_ssl_certificates_{database_type}")
+        def create_ssl_certificate(instance: str, connection_id: str) -> dict[str, Any]:
+            hook = CloudSQLHook(api_version="v1", gcp_conn_id=connection_id)
+            certificate_name = f"test_cert_{''.join(random.choice(string.ascii_letters) for _ in range(8))}"
+            response = hook.create_ssl_certificate(
+                instance=instance,
+                body={"common_name": certificate_name},
+                project_id=PROJECT_ID,
+            )
+            return response
+
+        create_ssl_certificate_task = create_ssl_certificate(
+            instance=cloud_sql_instance_name, connection_id=create_connection_task
+        )
+
+        @task(task_id=f"save_ssl_cert_locally_{database_type}")
+        def save_ssl_cert_locally(ssl_cert: dict[str, Any], db_type: str) -> dict[str, str]:
+            folder = SSL_COMPOSER_PATH_PREFIX if run_in_composer() else SSL_LOCAL_PATH_PREFIX
+            folder += f"/certs/{db_type}/{ssl_cert['operation']['name']}"
+            if not os.path.exists(folder):
```

Comment:
Can we simplify  `os.makedirs(folder, exist_ok=True)`?

Reply:
Makes sense, I have updated